src/modules/builders/builder/interface.nix

```diff
@@ -0,0 +1,37 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  t = lib.types;
+in {
+  options = {
+    disabled = lib.mkOption {
+      type = t.bool;
+      default = false;
+      description = "Whether to disable the builder, if disabled it can't be used.";
+    };
+    name = lib.mkOption {
+      type = t.str;
+      description = "Name of the builder.";
+    };
+    subsystem = lib.mkOption {
+      type = t.str;
+      description = "Subsystem of the builder.";
+    };
+    build = lib.mkOption {
+      type = t.functionTo (t.functionTo t.attrs);
+      default = _: _: {};
+    };
+    type = lib.mkOption {
+      type = t.enum [
+        "pure"
+        "ifd"
+      ];
+    };
+    version = lib.mkOption {
```

Comment:
I think builders don't need `version`. That field was specific to translators, as I needed to introduce this during a re-factoring.